src/rules/__tests__/no-deprecated-components.js

```diff
@@ -0,0 +1,34 @@
+const rule = require('../no-deprecated-components')
+const {RuleTester} = require('eslint')
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 'latest',
+    sourceType: 'module',
+    ecmaFeatures: {
+      jsx: true
+    }
+  }
+})
+
+ruleTester.run('no-deprecated-components', rule, {
+  valid: [`import {ActionList} from '@primer/react';`, `import {Tooltip} from '@primer/react/experimental';`],
+  invalid: [
+    {
+      code: `import {ActionList} from '@primer/react/deprecated';`,
+      errors: [
+        {
+          messageId: 'deprecatedComponent'
+        }
+      ]
+    },
+    {
+      code: `import {Tooltip} from '@primer/react';`,
```

Comment:
For this type of check, I am not sure if I can access to the component `@deprecated` annotation 🤔 Let me know if you know this is possible. 

Alternatively, we can create a list of these components here and read from that list and the downside of it that we need to update this list as we add `@deprecation` notice to the component or move them to the deprecated folder.